### PR TITLE
Bug 223: UI Tab Switching

### PR DESCRIPTION
**Pull request by sudipt1999**
I am not sure if not passing the props to  `<FixedSizeList></FixedSizeList>` would be right or not !
as I saw in the codebase of taskcluster also and https://www.npmjs.com/package/react-window.

### CODE REVIEW

File: ui/src/views/AwsProvisioner/ViewAwsWorkerType/index.jsx [hunk 1]
```diff
@@ -142,12 +149,15 @@ export default class ViewAwsWorkerType extends Component {
...
         )}
         {!error && !loading && currentTab === 2 && (
-          <Ec2ResourcesTable
-            onTerminateInstance={this.handleTerminateInstance}
-            workerType={awsProvisionerWorkerType}
-            awsState={awsProvisionerWorkerTypeState}
-            actionLoading={actionLoading}
-          />
+          <FixedSizeList>
+            <Spinner className={classes.spinner} {...tabContentLoader} />
```
Comment (helfi92) [inline ui/src/views/AwsProvisioner/ViewAwsWorkerType/index.jsx]: `FixedSizeList` should take a function as a child. See https://github.com/taskcluster/taskcluster/blob/7b058a76c4a92305c85083cfbcf19effcb53603a/ui/src/components/RoleScopesTable/index.jsx#L142 for an example. You can find examples and prop definitions in https://react-window.now.sh/#/api/FixedSizeList.

Comment (helfi92) [reply ui/src/views/AwsProvisioner/ViewAwsWorkerType/index.jsx]: I also think the `FixedSizeList` should be in the render method of `Ec2ResourcesTable`.

Comment (sudipt1999) [reply ui/src/views/AwsProvisioner/ViewAwsWorkerType/index.jsx]: I will do the needful!

**Comment by djmitche** [comment]
@sudipt1999 can you change the title of this PR to something more descriptive?

**Comment by sudipt1999** [comment]
@helfi92 kindly review it please!

**Comment by sudipt1999** [comment]
@helfi92 I am really sorry !
This one is from my link you mentioned in issue
Screenshot from 2019-03-26 18-38-34



This one is from my local environment!

Screenshot from 2019-03-26 18-31-30

I am not passing the scopes thats why I thought I was doing right!

**Comment by helfi92** [comment]
You need to login in order to view this page.

**Comment by sudipt1999** [comment]
@helfi92  I don't have the credentials for login into system!

**Comment by helfi92** [comment]
You'll need to use taskcluster-cli to get taskcluster credentails. The instructions for that can be found [here](https://github.com/taskcluster/taskcluster/tree/master/services/web-server#taskcluster-credentials). Let me know if you had questions.

**Comment by helfi92** [comment]
@sudipt1999 How is it going? Let me know if you had questions.

**Comment by sudipt1999** [comment]
Will update soon sir! Was stuck with doc update and other work ! Really sorry for delay!

**Comment by sudipt1999** [comment]
I tried getting credentials but wasn't successful
```javascript
defaults:
  taskcluster:
      rootUrl: https://taskcluster-web.netlify.com
  pulse:
    namespace:
    username: ..
    password: ..
    hostname: ..
    vhost: ..
```
Screenshot from 2019-04-13 17-37-53

I tried in different terminal also but isn't getting any thing!
I am going wrong somewhere but couldn't find it !

**Comment by sudipt1999** [comment]
I tried getting credentials but wasn't successful
```javascript
defaults:
  taskcluster:
      rootUrl: https://taskcluster-web.netlify.com
  pulse:
    namespace:
    username: ..
    password: ..
    hostname: ..
    vhost: ..
```
Screenshot from 2019-04-13 17-37-53

I tried in different terminal also but isn't getting any thing!
I am going wrong somewhere but couldn't find it !

**Comment by helfi92** [comment]
`eval $(./taskcluster signin --name taskcluster-web-server)` will open a browser tab where you will need to login. Once login is done, the credentials will be set for you.

**Comment by sudipt1999** [comment]
> Once login is done

For login only I require credentials actually sir !

**Comment by sudipt1999** [comment]
I logged in using this https://docs.taskcluster.net/docs/manual/design/apis/hawk/temporary-credentials and still it is not showing the credentials 
Screenshot from 2019-04-14 22-10-46

I am really sorry for delaying

**Comment by helfi92** [comment]
@sudipt1999 The steps for logging in are:

1. 
```bash
# This command will open a browser tab for you to login
eval $(./taskcluster signin --name taskcluster-web-server)
```

2. Click the "Sign In" button
Screen Shot 2019-04-15 at 8 43 32 AM

3. Once login is done, the credentials will be set.

Does that help?

**Comment by sudipt1999** [comment]
I logged in already
Screenshot from 2019-04-16 00-19-57


Still not wokring Sir!
Screenshot from 2019-04-16 00-20-13

**Comment by helfi92** [comment]
> eval $(./taskcluster signin --name taskcluster-web-server)

Running the `eval` command will open a window tab. You should **only** use this tab to login. You cannot manually go to tools.taskcluster.net to login.

The screenshot you uploaded does not show that you ran `eval $(./taskcluster signin --name taskcluster-web-server)`. The end result of the terminal will be like that:

Screen Shot 2019-04-15 at 3 19 20 PM

Screen Shot 2019-04-15 at 3 20 05 PM

**Comment by sudipt1999** [comment]
> > eval $(./taskcluster signin --name taskcluster-web-server)
> 
> Running the `eval` command will open a window tab. You should **only** use this tab to login. You cannot manually go to tools.taskcluster.net to login.
> 
> The screenshot you uploaded does not show that you ran `eval $(./taskcluster signin --name taskcluster-web-server)`. The end result of the terminal will be like that:
> 
> Screen Shot 2019-04-15 at 3 19 20 PM
> 
> Screen Shot 2019-04-15 at 3 20 05 PM

I actually did open the browser from the `eval $(.....` command only and did login also, I am not getting the message as your cli is showing 
> Credential output as environment variable

**Comment by helfi92** [comment]
Can you show me a screenshot of your terminal with the `eval` and also of what happens in the terminal after you login, please?

**Comment by sudipt1999** [comment]
Peek 2019-04-14 21-43

**Comment by helfi92** [comment]
Interesting. The spinner never goes away? Can you try to open the developer console/network tab to see if you see any errors.

**Comment by sudipt1999** [comment]
> Interesting. The spinner never goes away? Can you try to open the developer console/network tab to see if you see any errors.

The console doesn't show any error, sir!

**Comment by helfi92** [comment]
> Ahah there is CORBs error in the console log.

Could you post a screenshot of the CORS issue, please?

**Comment by sudipt1999** [comment]
ClientCreator.jsx:54 Cross-Origin Read Blocking (CORB) blocked cross-origin response https://auth.taskcluster.net/v1/clients/mozilla-auth0%2Fgithub%7C38929617%7Csudipt1999%2Ftaskcluster-web-server/reset with MIME type text/html. See https://www.chromestatus.com/feature/5629709824032768 for more details.

**Comment by helfi92** [comment]
@djmitche Any idea why @sudipt1999 would not be able to login via `taskcluster siginin`?

**Comment by djmitche** [comment]
That URL does not have content-type text/html.  I'm guessing there's a
proxy configured in the browser that is rewriting the response for some
reason.

**Comment by sudipt1999** [comment]
I solved the CORB issue by disabling web-security in chrome, 
now a error logs up in console while try to 
Screenshot from 2019-04-18 23-09-39

```
OPTIONS http://localhost:5080/graphql 
net::ERR_CONNECTION_REFUSED
```

I tried changing the web-server `config.yml` file and setting localhost port to **5080** but isn't working

**Comment by sudipt1999** [comment]
@helfi92  can you please help me out here, I am unable to fix this error on console please!

**Comment by helfi92** [comment]
> I tried changing the web-server config.yml file and setting localhost port to 5080 but isn't working

Please do not change `config.yml`. Create a file `user-config.yml` if it's no there in `services/web-server` with the following content:

```bash
defaults:
  # Configuration of access to other taskcluster components
  taskcluster:
    rootUrl: https://taskcluster.net
```

then you can do `yarn && yarn workspace taskcluster-web-server start` to start the server. Let me know if that helps.

**Comment by sudipt1999** [comment]
Screenshot from 2019-04-23 18-57-08


I am getting this!

**Comment by sudipt1999** [comment]
> > I tried changing the web-server config.yml file and setting localhost port to 5080 but isn't working
> 
> Please do not change `config.yml`. Create a file `user-config.yml` if it's no there in `services/web-server` with the following content:
> 
> ```shell
> defaults:
>   # Configuration of access to other taskcluster components
>   taskcluster:
>     rootUrl: https://taskcluster.net
> ```
> 
> then you can do `yarn && yarn workspace taskcluster-web-server start` to start the server. Let me know if that helps.

Even though on development the `3050` port should be accesses as given in docs but somehow it is accessing `5080` port for development also.

**Comment by helfi92** [comment]
@sudipt1999 Now that I think about it, I'm not sure if this issue is worth working on anymore. This whole view is intended to be deleted after once we have the new worker-manager (currently a WIP). I'll close the issue but feel free to grab another issue. You've been doing some great work!

**Pull request closed without merging**